extensions/wikia/Chat2/ChatAjax.class.php

```diff
@@ -22,129 +26,143 @@ class ChatAjax {
-			$stats = $userStatsService->getStats();
+			$stats = ( new UserStatsService( $user->getId() ) )->getStats();
 
-			// NOTE: This is attached to the user so it will be in the wiki's content language instead of wgLang (which it normally will).
-			$stats['edits'] = $wgContLang->formatNum( $stats['edits'] );
-			if ( empty( $stats['date'] ) ) {
-				// If the user has not edited on this wiki, don't show anything
-				$retVal['since'] = "";
-			} else {
-				// this results goes to chat server, which obiously has no user lang
-				// so we just return a short month name key - it has to be translated on client side
-				$date = getdate( wfTimestamp( TS_UNIX, $stats['date'] ) );
-				$retVal['since'] =  $date;
-			}
+			// this results goes to chat server, which obiously has no user lang
+			// so we just return a short month name key - it has to be translated on client side
+			$res['since'] = !empty( $stats['date'] )
+				? getdate( wfTimestamp( TS_UNIX, $stats['date'] ) )
+				: '';
 
-			$retVal['editCount'] = $stats['edits'];
+			// NOTE: This is attached to the user so it will be in the wiki's content language instead of wgLang (which it normally will).
+			$res['editCount'] = $wgContLang->formatNum( $stats['edits'] );
 		}
 
 		wfProfileOut( __METHOD__ );
-		return $retVal;
-	} // end getUserInfo()
+
+		return $res;
+	}
 
 	/**
 	 *  injecting data from chat to memcache
-	 *  and purging cache for ChatEntryPoint used by Anons
+	 *  and purging cache for ChatWidget used by Anons
 	 */
-
 	static public function setUsersList() {
-		ChatHelper::info( __METHOD__ . ': Method called' );
 		global $wgRequest;
+
 		wfProfileIn( __METHOD__ );
+		Chat::info( __METHOD__ . ': Method called' );
 
-		if ( ChatHelper::getChatCommunicationToken() != $wgRequest->getVal( 'token' ) ) {
+		if ( !self::authenticateServer() ) {
 			wfProfileOut( __METHOD__ );
-			return array( 'status' => false );
+
+			return [ 'status' => false ];
 		}
 
-		NodeApiClient::setChatters( $wgRequest->getArray( 'users' ) );
+		Chat::setChatters( $wgRequest->getArray( 'users' ) );
 
 		wfProfileOut( __METHOD__ );
-		return array( 'status' => $wgRequest->getArray( 'users' ) );
+
+		return [ 'status' => $wgRequest->getArray( 'users' ) ];
 	}
 
 	/**
-	 * Ajax endpoint for createing / accessing  private rooms
+	 * Ajax endpoint for creating / accessing  private rooms
 	 */
-
 	static public function getPrivateRoomID() {
-		ChatHelper::info( __METHOD__ . ': Method called' );
-		global $wgRequest;
+		global $wgRequest, $wgUser;
+
 		wfProfileIn( __METHOD__ );
+		Chat::info( __METHOD__ . ': Method called' );
+
+		if ( !self::authenticateServerOrUser() ) {
+			wfProfileOut( __METHOD__ );
+
+			return [ 'id' => false ];
+		}
 
 		$users = json_decode( $wgRequest->getVal( 'users' ) );
-		$roomId = NodeApiClient::getDefaultRoomId( 'private', $users );
+		if ( !$users || !is_array( $users ) || !in_array($wgUser->getName(), $users) ) {
```

Comment:
the `in_array` call needs proper spacing: `in_array( $wgUser->getName(), $users )`